loadtest/control/browsercontroller/controller.go

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package browsercontroller
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/control"
+	"github.com/mattermost/mattermost-load-test-ng/loadtest/user"
+)
+
+// TODO: Get this from the coordinator file
+const BROWSER_AGENT_API_URL = "BROWSER_AGENT_API_URL"
+
+// BrowserController is a controller that manages browser sessions
+// by communicating with the browserLt API server.
+type BrowserController struct {
+	id            int
+	user          user.User
+	status        chan<- control.UserStatus
+	rate          float64
+	stopChan      chan struct{}
+	stoppedChan   chan struct{}
+	wg            *sync.WaitGroup
+	browserAPIURL string
+	httpClient    *http.Client
+	isRunning     bool
+	mu            sync.Mutex
+}
+
+// AddBrowserRequest represents the request body for adding a browser session
+type AddBrowserRequest struct {
+	UserID   string `json:"userId"`
+	Password string `json:"password"`
+}
+
+// RemoveBrowserRequest represents the request body for removing a browser session
+type RemoveBrowserRequest struct {
+	UserID string `json:"userId"`
+}
+
+// BrowserAPIResponse represents the response from the browser API
+type BrowserAPIResponse struct {
+	Success bool   `json:"success"`
+	Message string `json:"message"`
+	Error   *struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+	} `json:"error,omitempty"`
+}
+
+// New creates and initializes a new BrowserController with given parameters.
+func New(id int, user user.User, status chan<- control.UserStatus) (*BrowserController, error) {
+	if user == nil {
+		return nil, errors.New("user cannot be nil")
+	}
+	if status == nil {
+		return nil, errors.New("status channel cannot be nil")
+	}
+
+	// Get browser API URL from environment variable (same as browser server uses)
+	browserAPIURL := os.Getenv("BROWSER_AGENT_API_URL")
+	if browserAPIURL == "" {
+		return nil, errors.New("BROWSER_AGENT_API_URL environment variable is required")
+	}
```

Comment:
I understand we will be getting this from the config, right?

Reply:
implementation is now changed, remove this getenv